### PR TITLE
AR-847 filter lessons by likutim

### PR DESCRIPTION
**Pull request by davgur**

### CODE REVIEW

File: api/handlers.go [hunk 1]
```diff
@@ -2314,6 +2320,33 @@ func appendContentTypesFilterMods(mods *[]qm.QueryMod, f ContentTypesFilter) err
...
+		}
+		a[i] = ct.ID
+	}
+	q := `id IN (
+		SELECT cud.source_id FROM content_units cu 
+		INNER JOIN content_unit_derivations cud ON cu.id = cud.derived_id 
+		WHERE cu.type_id = ?
+	)`
+	*mods = append(*mods,
+		qm.WhereIn(q, a...))
```
Comment (edoshor) [inline api/handlers.go:2345]: could be one line

File: api/handlers.go [hunk 2]
```diff
@@ -2314,6 +2320,33 @@ func appendContentTypesFilterMods(mods *[]qm.QueryMod, f ContentTypesFilter) err
...
+		return nil
+	}
+
+	a := make([]interface{}, len(f.DerivedTypes))
+	for i, x := range f.DerivedTypes {
+		ct, ok := mdb.CONTENT_TYPE_REGISTRY.ByName[strings.ToUpper(x)]
+		if !ok {
+			return errors.Errorf("Unknown derive content type: %s", x)
+		}
+		if _, p := consts.PERMITTED_UNIT_CT_FOR_DERIVED_FILTER[ct.Name]; !p {
```
Comment (edoshor) [inline api/handlers.go:2334]: maybe rename `p` to `ok`? 
as far as I know it does not conflict with the `ok` from above it as it is only declared in the `if` statement scope.

**Review by edoshor — APPROVED** [review]
Looks good. 2 small style comments (not mandatory to fix)

File: api/handlers.go (hunk 1 above)
Comment (davgur) [reply api/handlers.go]: fixed

File: api/handlers.go (hunk 2 above)
Comment (davgur) [reply api/handlers.go]: fixed

**Pull request merged**